### PR TITLE
[Feature] support reading viewfs://

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [X] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by DorianZheng** [comment]
plz add ut

**Comment by dirtysalt** [comment]
> plz add ut

viewfs is system level coniguration, and can not be written in ut.

File: be/src/fs/fs.cpp [hunk 1]
```diff
@@ -56,11 +56,11 @@ inline bool starts_with(std::string_view s, std::string_view prefix) {
 
 inline bool is_s3_uri(std::string_view uri) {
     return starts_with(uri, "oss://") || starts_with(uri, "s3n://") || starts_with(uri, "s3a://") ||
-           starts_with(uri, "s3://");
+           starts_with(uri, "s3://") || starts_with(uri, "cos://");
```
Comment (stephen-shelby) [inline be/src/fs/fs.cpp:59]: Add some ut? Avoid other students from missing some schemas in the next refactoring

Comment (stephen-shelby) [inline be/src/fs/fs.cpp:59]: Add some ut for this function? Avoid other students from missing some schemas in the next refactoring

**Review by DorianZheng — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.2

**Pull request merged**